src/machine/machine.c

```diff
@@ -523,6 +524,36 @@ int machine_finalize(Machine *m) {
         return 0;
 }
 
+static bool machine_validate_unit(Machine *m) {
+        int r;
+        _cleanup_free_ char *unit = NULL;
+        _cleanup_free_ char *cgroup = NULL;
+
+        r = cg_pid_get_unit(m->leader, &unit);
+        if (!r && streq(m->unit, unit))
+                return true;
+
+        if (r == -ESRCH)
+                log_info("Machine unit is inactive, but the leader process is exited. "
+                        "machine: %s, leader: "PID_FMT", unit: %s.", m->name, m->leader, m->unit);
+        else if (r)
+                log_info_errno(r, "Can not get unit from cgroup. "
+                        "machine: %s, leader: "PID_FMT", unit: %s, error: %m", m->name, m->leader, m->unit);
+        else if (unit && !streq(m->unit, unit))
+                log_info("Machine unit name not match. "
+                        "machine: %s, leader: "PID_FMT", machine unit: %s, real unit: %s", m->name, m->leader, m->unit, unit);
+
+        r = manager_get_unit_cgroup_path(m->manager, m->unit, &cgroup);
+        if (!r && !isempty(cgroup) && cg_is_empty_recursive(SYSTEMD_CGROUP_CONTROLLER, cgroup) > 0) {
+                log_info("Cgroup is empty in the machine unit. "
+                        "machine: %s, leader: "PID_FMT", machine unit: %s.", m->name, m->leader, m->unit);
+                /*The vm will be added to gc list only when there is no any process in the scope*/
+                return false;
```

Comment:
Here too, no boolean comparisons for `r` and the message should be one line.

Reply:
ok.